tools/eslint-rules/snake-case-filenames.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @fileoverview enforces snake_case for filenames.
+ * @author Shobhit Chittora <chittorashobhit@gmail.com>
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const path = require('path');
+
+module.exports = function(context) {
+  const snakeCaseRegex = new RegExp('^[a-z_]+$');
```

Comment:
Nit: Probably more idiomatic and consistent with the rest of our code base to not use the constructor here:

```js
  const snakeCaseRegex = /^[a-z_]+$/;
```